snapcraft/plugins/catkin.py

```diff
@@ -348,36 +348,65 @@ def _prepare_build(self):
         # dependencies contain .cmake files pointing to system paths (e.g.
         # /usr/lib, /usr/include, etc.). They need to be rewritten to point to
         # the install directory.
-        def rewrite_paths(match):
+        def _new_path(path):
+            if not path.startswith(self.installdir):
+                return self.installdir + path
```

Comment:
Note that it's [guaranteed to be absolute](https://github.com/snapcore/snapcraft/pull/1130/files#diff-3cc26b49a70ee86523598a96dd385301R363).